src/structural_transformation/symbolics_tearing.jl

```
@@ -3,185 +3,122 @@ function tearing_sub(expr, dict, s)
     s ? simplify(expr) : expr
 end
 
-function tearing_reassemble(sys; simplify=false)
+function tearing_reassemble(sys, var_eq_matching; simplify=false)
     s = structure(sys)
-    @unpack fullvars, partitions, var_eq_matching, graph, scc = s
+    @unpack fullvars, solvable_graph, graph = s
+
     eqs = equations(sys)
 
     ### extract partition information
-    rhss = []
-    solvars = []
-    ns, nd = nsrcs(graph), ndsts(graph)
-    active_eqs  = trues(ns)
-    active_vars = trues(nd)
-    rvar2reqs = Vector{Vector{Int}}(undef, nd)
-    for (ith_scc, partition) in enumerate(partitions)
-        @unpack e_solved, v_solved, e_residual, v_residual = partition
-        for ii in eachindex(e_solved)
-            ieq = e_solved[ii]; ns -= 1
-            iv = v_solved[ii]; nd -= 1
-            rvar2reqs[iv] = e_solved
-
-            active_eqs[ieq] = false
-            active_vars[iv] = false
-
-            eq = eqs[ieq]
-            var = fullvars[iv]
-            rhs = value(solve_for(eq, var; simplify=simplify, check=false))
-            # if we don't simplify the rhs and the `eq` is not solved properly
-            (!simplify && occursin(rhs, var)) && (rhs = SymbolicUtils.polynormalize(rhs))
+    function solve_equation(ieq, iv)
+        var = fullvars[iv]
+        eq = eqs[ieq]
+        rhs = value(solve_for(eq, var; simplify=simplify, check=false))
+
+        if var in vars(rhs)
```

This should just throw an internal error.

---

Ok. Let's do that in a separate PR though, since that's a behavior change?